### DISCUSSION TITLE
Forecasting Horizon when calling the fit method

### DISCUSSION THREAD

**Discussion by edzWH** [alan-turing-institute/sktime#5955] [Q&A] [answered]
If I'm calling the fit method after doing some CV on a pipeline, is the forecasting horizon irrelevant if I'm purely using the object to document the best params/scores and not fit a model?

For additional context, here is the code I'm running:

` gscv = ForecastingRandomizedSearchCV(forecaster, cv=cv, param_distributions=param_grid, n_iter=random_grid, refit=True)
    
        result = gscv.fit(y=y_train, X=X_train,fh=fh)`

After the random grid search has finished, I'm not looking to fit the model to future data, I'm just creating a dict out of the results to save elsewhere. As I'm not forecasting in this instance can the forecasting horizon just be a minimum length array? Or should the FH be consistent with the test length in the CV? 

Thanks.

**Comment by fkiraly** [comment, marked as answer]
> If I'm calling the fit method after doing some CV on a pipeline, is the forecasting horizon irrelevant if I'm purely using the object to document the best params/scores and not fit a model?

Yes - in this case the `fh` in `fit` is unimportant, it is ignored in the internal benchmarking. What matters is the `fh` in, e.g., `SlidingWindowSplitter`, this is not automatically set to coincide with the `fh` you use later for a forecast.

If you only want to run a benchmarking experiment - and not make an additional forecast using the best parameter, you may want to consider using `evaluate` (from `forecasting.model_evaluation`), this is also used inside the grid search estimator.
